x-pack/plugins/observability_solution/slo/public/pages/slo_details/components/header_control.tsx

```diff
@@ -65,10 +75,15 @@ export function HeaderControl({ isLoading, slo }: Props) {
   };
 
   const handleNavigateToRules = async () => {
-    const locator = locators.get<RulesParams>(rulesLocatorID);
+    if (rules.length === 1) {
```

Comment:
@shahzad31 This part is almost a duplicate with `x-pack/plugins/observability_solution/slo/public/pages/slos/components/slo_item_actions.tsx`. Could these 2 files share a common `handleNavigateToRules`?

Reply:
this duplication persists in all of the actions, so perhaps we can do a refactor in a follow up to combine actions component